### PR TITLE
TST: Split and simplify test_value_counts_unique_nunique

### LINKED ISSUE #32205: Simplify test_value_counts_unique_nunique_null in pandas/tests/base/test_ops.py
As stated by @jreback in [his comment](https://github.com/pandas-dev/pandas/pull/32046#discussion_r383021295) on #32046, there are some weird constructs, duplications and unreachable code in this test.

For more info see the mentioned PR and the unresolved comments in the bottom.

The other tests in this file have a similar problem, but should probably be a separate PR.

### LINKED ISSUE #32220: CI: value_counts tests failing sometimes
It seems like we're getting some failures from `test_value_counts_unique_nunique` in `pandas/tests/base/test_ops.py` (weird thing is that it only fails some of the time; maybe because of randomized test data?). Looks like these tests were updated in https://github.com/pandas-dev/pandas/pull/32046; @SaturnFromTitan any ideas what might be happening there?

Sample failed pipeline: https://github.com/pandas-dev/pandas/runs/464853016

### PR DESCRIPTION
**Pull request by SaturnFromTitan**
closes #32205, closes #32220

- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by SaturnFromTitan** [comment]
The failing CI is unrelated to these changes

File: pandas/tests/base/test_ops.py [hunk 1]
```diff
@@ -205,63 +222,44 @@ def test_ndarray_compat_properties(self, index_or_series_obj):
...
+        expected = pd.Series(dict(counter.most_common()), dtype=np.int64)
+        expected.index = expected.index.astype(obj.dtype)
+        if isinstance(obj, pd.MultiIndex):
+            expected.index = pd.Index(expected.index)
 
-        # dropna=True would break for MultiIndex
-        assert obj.nunique(dropna=False) == len(np.unique(obj.values))
+        # sort_index to avoid switched order when values share the same count
+        result = result.sort_index()
+        expected = expected.sort_index()
```
Comment (jreback) [inline pandas/tests/base/test_ops.py]: can you split the test right below as well (alternatively maybe able to combine with the above as they are duplicative, but no big deal)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/base/test_ops.py (hunk 1 above)
Comment (SaturnFromTitan) [reply pandas/tests/base/test_ops.py]: @jreback Done - please have another look. I think the `test_x_null` cases are sufficiently different to not merge them with the non-null ones.

**Comment by SaturnFromTitan** [comment]
@jreback merged master to resolve conflicts

File: pandas/conftest.py [hunk 2]
```diff
@@ -1015,7 +1015,7 @@ def _create_series(index):
     """ Helper for the _series dict """
     size = len(index)
     data = np.random.randn(size)
-    return pd.Series(data, index=index, name="a")
+    return pd.Series(data, index=index)
```
Comment (jreback) [inline pandas/conftest.py]: why remove this?

File: pandas/tests/base/test_ops.py [hunk 3]
```diff
@@ -205,176 +204,152 @@ def test_ndarray_compat_properties(self, index_or_series_obj):
...
+        obj = index_or_series_obj
+        obj = np.repeat(obj, range(1, len(obj) + 1))
+        result = obj.unique()
 
-        # create repeated values, 'n'th element is repeated by n+1 times
-        if isinstance(obj, Index):
-            expected_index = Index(obj[::-1])
-            expected_index.name = None
-            obj = obj.repeat(range(1, len(obj) + 1))
+        # dict.fromkeys preserves the order
```
Comment (jreback) [inline pandas/tests/base/test_ops.py]: why do you need dict.fromkeys?

File: pandas/tests/base/test_ops.py [hunk 4]
```diff
@@ -205,176 +204,152 @@ def test_ndarray_compat_properties(self, index_or_series_obj):
...
-        expected_s = Series(
-            range(len(orig), 0, -1), index=expected_index, dtype="int64"
-        )
+            expected = np.array(unique_values)
+            tm.assert_numpy_array_equal(result, expected)
 
-        result = obj.value_counts()
-        tm.assert_series_equal(result, expected_s)
-        assert result.index.name is None
+    @pytest.mark.parametrize("null_obj", [np.nan, None])
```
Comment (jreback) [inline pandas/tests/base/test_ops.py]: can you use nulls_fixture

File: pandas/tests/base/test_ops.py [hunk 5]
```diff
@@ -205,176 +204,152 @@ def test_ndarray_compat_properties(self, index_or_series_obj):
...
+        assert obj.nunique(dropna=False) == expected
 
     @pytest.mark.parametrize("null_obj", [np.nan, None])
-    def test_value_counts_unique_nunique_null(self, null_obj, index_or_series_obj):
-        orig = index_or_series_obj
-        obj = orig.copy()
-        klass = type(obj)
-        values = obj._ndarray_values
-        num_values = len(orig)
+    def test_nunique_null(self, null_obj, index_or_series_obj):
```
Comment (jreback) [inline pandas/tests/base/test_ops.py]: same

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/base/test_ops.py (hunk 3 above)
Comment (SaturnFromTitan) [reply pandas/tests/base/test_ops.py]: It seemed like the easiest way to get the unique values of obj while preserving the order. I'm open to alternatives though

File: pandas/tests/base/test_ops.py (hunk 4 above)
Comment (SaturnFromTitan) [reply pandas/tests/base/test_ops.py]: I tried using `nulls_fixture` and `unique_nulls_fixture`. Nearly all tests break for `pd.NaT` and `pd.NA` though...

File: pandas/conftest.py (hunk 2 above)
Comment (SaturnFromTitan) [reply pandas/conftest.py]: No good reason, I'll undo it

File: pandas/tests/base/test_ops.py (hunk 4 above)
Comment (jreback) [reply pandas/tests/base/test_ops.py]: ok, yeah we need to test these, can you create an issue. we will want to add these even if they need xfailing for now as there is no testing on them.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @SaturnFromTitan 

one other followup on this. I think we should split out the tests you just changed (e.g. unique, value_counts) into a test_unique.py module; happy to do this generally as well (though should be separate from a refactorting PR).

File: pandas/tests/base/test_ops.py (hunk 4 above)
Comment (SaturnFromTitan) [reply pandas/tests/base/test_ops.py]: xref: #32437

**Comment by SaturnFromTitan** [comment]
Will do. I'll wait until #32311 is merged though as I think it should go into the test_uniques.py as well

**Pull request merged**